pkg/descheduler/strategies/topologyspreadconstraint.go

```diff
@@ -111,6 +112,12 @@ func RemovePodsViolatingTopologySpreadConstraint(
 				if constraint.WhenUnsatisfiable == v1.ScheduleAnyway && (strategy.Params == nil || !strategy.Params.IncludeSoftConstraints) {
 					continue
 				}
+				// Need to check v1.TopologySpreadConstraint deepEquality because
+				// v1.TopologySpreadConstraint.LabelSelector is a pointer
+				// and assigning new constrains would always add more keys
+				if hasIdenticalConstraints(constraint, namespaceTopologySpreadConstraints) {
```

Comment:
At this point, why even use a map? we can just convert to a slice

Reply:
that's true...

What if we made the label selector be always the same, but only changing its contents, like with:
```
func setBaseLabelSelector(constraint v1.TopologySpreadConstraint, baseLabelSelector *metav1.LabelSelector ) v1.TopologySpreadConstraint {
	baseLabelSelector.MatchExpressions = constraint.LabelSelector.MatchExpressions
	baseLabelSelector.MatchLabels = constraint.LabelSelector.MatchLabels
	constraint.LabelSelector = baseLabelSelector
	return constraint
}
```

Reply:
I like your approach of `DeepEqual` because soon we will have to add support for [MinDomains](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/3022-min-domains-in-pod-topology-spread) and it would just work. And later down the road additional fields will be added and with the current implementation, we won't have to make any additional changes.

Reply:
@knelasevero label selectors of two constraints do not have to be always equal. You can have two constraints in the same namespace just with different label selectors. E.g. color=red and color=blue.

Reply:
@a7i ok, makes sense!

@ingvagabund , but in that code snippet I'm grabbing the inner values that are different. I'm just keeping that "same label selector" object so it has the same address, but the values are changing to the ones that come from the constraint.

Reply:
Changed it to slice, let me know what you think

Reply:
Would not all the constraints have the same label selector equal to the label selector of the last constraint given all the constraint share the same address of the baseLabelSelector?